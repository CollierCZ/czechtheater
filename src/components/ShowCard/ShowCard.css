/* Cards */

.showList {
    display: -moz-flex;
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    -moz-flex-wrap: wrap;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    position: relative;
    margin: -2.5em 0 1.5rem -2.5em;
}

    .showList .showCard {
        -moz-transition: -moz-transform 0.5s ease, opacity 0.5s ease;
        -webkit-transition: -webkit-transform 0.5s ease, opacity 0.5s ease;
        -ms-transition: -ms-transform 0.5s ease, opacity 0.5s ease;
        transition: transform 0.5s ease, opacity 0.5s ease;
        position: relative;
        width: calc(33.3333333333% - 2.5em);
        margin: 2.5em 0 0 2.5em;
    }

        .showList .showCard > .image {
            -moz-transition: -moz-transform 0.5s ease;
            -webkit-transition: -webkit-transform 0.5s ease;
            -ms-transition: -ms-transform 0.5s ease;
            transition: transform 0.5s ease;
            position: relative;
            display: block;
            width: 100%;
            border-radius: 4px;
            overflow: hidden;
        }

            .showList .showCard > .image img {
                display: block;
                width: 100%;
                margin-bottom: 0;
            }

            .showList .showCard > .image:before {
                pointer-events: none;
                -moz-transition: background-color 0.5s ease, opacity 0.5s ease;
                -webkit-transition: background-color 0.5s ease, opacity 0.5s ease;
                -ms-transition: background-color 0.5s ease, opacity 0.5s ease;
                transition: background-color 0.5s ease, opacity 0.5s ease;
                content: '';
                display: block;
                position: absolute;
                top: 0;
                left: 0;
                width: 100%;
                height: 100%;
                opacity: 1.0;
                z-index: 1;
                opacity: 0.8;
            }

        .showList .showCard > a {
            display: -moz-flex;
            display: -webkit-flex;
            display: -ms-flex;
            display: flex;
            -moz-flex-direction: column;
            -webkit-flex-direction: column;
            -ms-flex-direction: column;
            flex-direction: column;
            -moz-align-items: center;
            -webkit-align-items: center;
            -ms-align-items: center;
            align-items: center;
            -moz-justify-content: center;
            -webkit-justify-content: center;
            -ms-justify-content: center;
            justify-content: center;
            -moz-transition: background-color 0.5s ease, -moz-transform 0.5s ease;
            -webkit-transition: background-color 0.5s ease, -webkit-transform 0.5s ease;
            -ms-transition: background-color 0.5s ease, -ms-transform 0.5s ease;
            transition: background-color 0.5s ease, transform 0.5s ease;
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            padding: 1em;
            border-radius: 4px;
            border-bottom: 0;
            color: #ffffff;
            text-align: center;
            text-decoration: none;
            z-index: 3;
        }

            .showList .showCard > a > :last-child {
                margin: 0;
            }

            .showList .showCard > a:hover {
                color: #ffffff !important;
            }

            .showList .showCard > a h3 {
                margin: 0;
                font-variant-caps: all-small-caps;
            }

            .showList .showCard > a .showDescription {
                -moz-transition: max-height 0.5s ease, opacity 0.5s ease;
                -webkit-transition: max-height 0.5s ease, opacity 0.5s ease;
                -ms-transition: max-height 0.5s ease, opacity 0.5s ease;
                transition: max-height 0.5s ease, opacity 0.5s ease;
                width: 100%;
                max-height: 0;
                line-height: 1.25;
                font-size: 0.9em;
                margin-top: 0.35em;
                opacity: 0;
            }

                .showList .showCard > a .showDescription > :last-child {
                    margin-bottom: 0;
                }

        .showList .showCard:nth-child(3n+1) > .image:before {
            background-color: #11457e;
        }
        
        .showList .showCard:nth-child(3n+2) > .image:before {
            background-color: #000000;
        }

        .showList .showCard:nth-child(3n+3) > .image:before {
            background-color: #d7141a;
        }

        body:not(.is-touch) .showList .showCard:hover > .image {
            -moz-transform: scale(1.1);
            -webkit-transform: scale(1.1);
            -ms-transform: scale(1.1);
            transform: scale(1.1);
        }

            body:not(.is-touch) .showList .showCard:hover > .image:before {
                background-color: #333333;
                opacity: 0.35;
            }

            body:not(.is-touch) .showList .showCard:hover > .image:after {
                opacity: 0;
            }

        body:not(.is-touch) .showList .showCard:hover .showDescription {
            max-height: 15em;
            opacity: 1;
        }

    body.is-preload .showList .showCard {
        -moz-transform: scale(0.9);
        -webkit-transform: scale(0.9);
        -ms-transform: scale(0.9);
        transform: scale(0.9);
        opacity: 0;
    }

    body.is-touch .showList .showCard .showDescription {
        max-height: 15em;
        opacity: 1;
    }

    @media screen and (max-width: 1280px) {

        .showList {
            margin: -1.25em 0 1.25rem -1.25em;
        }

            .showList .showCard {
                width: calc(33.3333333333% - 1.25em);
                margin: 1.25em 0 0 1.25em;
            }

    }

    @media screen and (max-width: 980px) {

        .showList {
            margin: -2.5em 0 1.5rem -2.5em;
        }

            .showList .showCard {
                width: calc(50% - 2.5em);
                margin: 2.5em 0 0 2.5em;
            }

    }

    @media screen and (max-width: 736px) {

        .showList {
            margin: -1.25em 0 1rem -1.25em;
        }

            .showList .showCard {
                width: calc(50% - 1.25em);
                margin: 1.25em 0 0 1.25em;
            }

                .showList .showCard:hover > .image {
                    -moz-transform: scale(1.0);
                    -webkit-transform: scale(1.0);
                    -ms-transform: scale(1.0);
                    transform: scale(1.0);
                }

    }

    @media screen and (max-width: 480px) {

        .showList {
            margin: 0;
            margin-bottom: 1rem;
        }

            .showList .showCard {
                width: 100%;
                margin: 1.25em 0 0 0;
            }

    }

    .showList .showCard > .tickets + a {
        height: 79%;
    }