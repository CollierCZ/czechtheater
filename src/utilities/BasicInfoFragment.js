import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


