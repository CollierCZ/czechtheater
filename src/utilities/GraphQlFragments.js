import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


