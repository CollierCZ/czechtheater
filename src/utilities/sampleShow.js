export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }