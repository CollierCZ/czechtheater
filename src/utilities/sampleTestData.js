export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge"
    },
    description: {
      resolvedHtml: "<p>All the acting</p>"
    },
    main_image: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    },
    ticket_link: {
      value: "https://example.com"
    },
    url: {
      value: "alquists-revenge"
    },
    gallery: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    }
  }
}

export const sampleShowNoGallery = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.gallery.assets = []

export const sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.ticket_link.value = ""

export const basicInfoQuery = {
  kenticoCloudItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater"
      },
      favicon: {
        assets: [{
          url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg"
        }]
      },
      short_description: {
        value: "A great theater."
      }
    }
  }
}