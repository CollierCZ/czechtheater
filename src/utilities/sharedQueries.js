import { graphql } from "gatsby";

export const basicInfoQuery = graphql`
  query BasicInfoQuery {
    kenticoCloudItemBasicInfo {
      elements {
        name {
          value
        }
        short_description {
          value
        }
        favicon {
          assets{
            url
          }
        }
      }
    }
  }
`