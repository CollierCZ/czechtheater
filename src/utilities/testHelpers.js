import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}