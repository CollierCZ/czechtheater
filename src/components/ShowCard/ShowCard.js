import React, { Component } from "react";
import { ImageElement } from "@kentico/gatsby-kontent-components";
import "./ShowCard.css";

class ShowCard extends Component {
  render() {
    const {
      name,
      description,
      showImage,
      slug,
      time,
      ticketLink,
    } = this.props;
    return (
      <div className={time + " showCard"}>
        <div className="image">
          <ImageElement image={showImage} alt={showImage.description} layout="fixed" width={291} height={291} />
        </div>
        {ticketLink
          ? time === "future" && (
              <h4 className="tickets">
                <a href={ticketLink}>Tickets</a>
              </h4>
            )
          : null}
        <a href={slug}>
          <h3>{name}</h3>
          <div
            className="showDescription"
            dangerouslySetInnerHTML={{ __html: description }}
          />
        </a>
      </div>
    );
  }
}

export default ShowCard;
