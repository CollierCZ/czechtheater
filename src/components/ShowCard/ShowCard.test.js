import React from "react";
import ShowCard from "./ShowCard";
import { simpleShallowRender } from "../../utilities/testHelpers"

const propsArray = {
  name: "Alquist's revenge",
  description: "All the acting",
  imageUrl: "https://example.com",
  imageDescription: "A cool image",
  slug: "alquists-revenge",
}

const ticketLink = "https://example.com"

describe("ShowCard", () => {
  it("loads correctly for future shows", () => {
    simpleShallowRender(<ShowCard {...propsArray} time="future" />)
  })
  it("loads correctly for past shows", () => {
    simpleShallowRender(<ShowCard {...propsArray} time="past" />)
  })
  it("loads correctly with a ticket link", () => {
    simpleShallowRender(<ShowCard {...propsArray} ticketLink={ticketLink} time="future" />)
  })
})