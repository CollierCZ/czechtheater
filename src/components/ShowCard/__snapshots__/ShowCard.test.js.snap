// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCard loads correctly for future shows 1`] = `
<div
  className="future showCard"
>
  <div
    className="image"
  >
    <img
      alt="A cool image"
      src="https://example.com?w=353&h=353&fit=crop"
    />
  </div>
  <a
    href="alquists-revenge"
  >
    <h3>
      Alquist's revenge
    </h3>
    <div
      className="showDescription"
      dangerouslySetInnerHTML={
        Object {
          "__html": "All the acting",
        }
      }
    />
  </a>
</div>
`;

exports[`ShowCard loads correctly for past shows 1`] = `
<div
  className="past showCard"
>
  <div
    className="image"
  >
    <img
      alt="A cool image"
      src="https://example.com?w=353&h=353&fit=crop"
    />
  </div>
  <a
    href="alquists-revenge"
  >
    <h3>
      Alquist's revenge
    </h3>
    <div
      className="showDescription"
      dangerouslySetInnerHTML={
        Object {
          "__html": "All the acting",
        }
      }
    />
  </a>
</div>
`;

exports[`ShowCard loads correctly with a ticket link 1`] = `
<div
  className="future showCard"
>
  <div
    className="image"
  >
    <img
      alt="A cool image"
      src="https://example.com?w=353&h=353&fit=crop"
    />
  </div>
  <h4
    className="tickets"
  >
    <a
      href="https://example.com"
    >
      Tickets
    </a>
  </h4>
  <a
    href="alquists-revenge"
  >
    <h3>
      Alquist's revenge
    </h3>
    <div
      className="showDescription"
      dangerouslySetInnerHTML={
        Object {
          "__html": "All the acting",
        }
      }
    />
  </a>
</div>
`;
